lib/ansible/modules/packaging/language/pip.py

```diff
@@ -521,11 +639,11 @@ def main():
         if extra_args:
             cmd += ' %s' % extra_args
 
+        cmd += ' '
         if name:
-            for pkg in name:
-                cmd += ' %s' % _get_full_name(pkg, version)
+            cmd += ' '.join('"%s"' % d for d in distributions)
         elif requirements:
-            cmd += ' -r %s' % requirements
+            cmd += '-r "%s"' % requirements
```

Comment:
Make sense, but open another pr for this is better since all operations on `cmd` in pip module use pattern like `cmd = xxx + xxx`, it needs many changes at many places, and it's not relevant to version specifier problem.